bin/fuel-core/src/cli/run.rs

```diff
@@ -510,10 +510,35 @@ pub async fn get_service_with_shutdown_listeners(
     info!("Fuel Core version v{}", env!("CARGO_PKG_VERSION"));
     trace!("Initializing in TRACE mode.");
     // initialize the server
+    #[cfg(feature = "rocksdb")]
+    let mut combined_database =
+        CombinedDatabase::from_config(&config.combined_db_config)?;
+
+    #[cfg(not(feature = "rocksdb"))]
     let combined_database = CombinedDatabase::from_config(&config.combined_db_config)?;
 
     let mut shutdown_listener = ShutdownListener::spawn();
 
+    // TODO: Handle graceful shutdown of migration processes
+    #[cfg(feature = "rocksdb")]
+    {
+        let on_chain_migration_handle = combined_database
```

Comment:
I think the `HistoricalRocksDB` should be responsible for launching the migration on construction.